src/components/metrics.tsx

```diff
@@ -84,6 +89,126 @@ import {
 // Stores information about the currently focused query input
 let focusedQuery;
 
+type PredefinedQueryType = {
+  name: string;
+  query: string;
+};
+
+export const PreDefinedQueriesDropdown = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState('');
+
+  const { t } = useTranslation('plugin__monitoring-plugin');
+
+  const predefinedQueries: PredefinedQueryType[] = [
+    {
+      name: 'CPU Usage',
+      // eslint-disable-next-line max-len
+      query: `sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate) by (pod)`,
+    },
+    {
+      name: 'Memory Usage',
+      // eslint-disable-next-line max-len
+      query: `sum(container_memory_working_set_bytes{container!=""}) by (pod)`,
+    },
+    {
+      name: 'Filesystem Usage',
+      // eslint-disable-next-line max-len
+      query: `topk(25, sort_desc(sum(pod:container_fs_usage_bytes:sum{container="",pod!=""}) BY (pod, namespace)))`,
+    },
+    {
+      name: 'Recieve bandwidth',
+      query: `sum(irate(container_network_receive_bytes_total[2h])) by (pod)`,
+    },
+    {
+      name: 'Transmit bandwidth',
+      query: `sum(irate(container_network_transmit_bytes_total[2h])) by (pod)`,
+    },
+    {
+      name: 'Rate of received packets',
+      query: `sum(irate(container_network_receive_packets_total[2h])) by (pod)`,
+    },
+    {
+      name: 'Rate of transmitted packets',
+      query: `sum(irate(container_network_transmit_packets_total[2h])) by (pod)`,
+    },
+    {
+      name: 'Rate of received packets dropped',
+      query: `sum(irate(container_network_receive_packets_dropped_total[2h])) by (pod)`,
+    },
+    {
+      name: 'Rate of transmitted packets dropped',
+      query: `sum(irate(container_network_transmit_packets_dropped_total[2h])) by (pod)`,
+    },
+  ];
+
+  // Note this fires twice when <Select> is clicked.
+  const onToggle = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const dispatch = useDispatch();
+
+  const queriesList = useSelector(({ observe }: RootState) =>
+    observe.getIn(['queryBrowser', 'queries']),
+  );
+
+  const insertPredefinedQuery = (query: string) => {
+    const queriesListDetails = queriesList.toJS();
+    const isInitialQueryEmpty =
+      queriesList.size === 1 &&
+      (queriesListDetails[0]?.text === '' ||
+        queriesListDetails[0]?.text === null ||
+        queriesListDetails[0]?.text === undefined);
+    const index = isInitialQueryEmpty ? 0 : queriesList.size;
```

Comment:
we might want to insert the query only if its different from the current first

Reply:
Updated in the latest commit.